### DISCUSSION TITLE
What is planned for the upcoming Engram 3.0?

### DISCUSSION THREAD

**Discussion by sunaku** [binarybottle/engram-layout#5] [Q&A] [answered]
Hello,

I'm very excited about the Engram layout (many thanks for creating it and explaining the thoughtful, systematic approach leading to its invention) and I'm eager to adopt it as my primary daily driver. :star_struck:  However, it seems the rate of development is rather fast paced because there is already a new 3.0 version being developed merely _days_ after the current 2.0 version had been finalized.

Would you recommend that I adopt the current 2.0 version now, or wait for the upcoming 3.0 version?  And what's planned for 3.0?

Thanks for your consideration.

**Comment by binarybottle** [comment]
Thank you for your interest, Suraj!

I have learned a lot very rapidly from my experiments as well as from feedback from the keyboard optimization community.  I am very close to a solution that I will be satisfied with, so please hold on a little longer!

Cheers,
@rno

**Reply by sunaku** [reply]
Regarding section "Assign letters Z and Q to keys 112 and 113 and test left/right swap" in commit 4516e5900bc920bb947c82acd0628997231f12cb:

Is the choice to keep Q and U together a personal preference? :thinking:  It's not clear why the top one isn't chosen.

```
All of the original left/right layouts are tied for top rank, with either J or Q in the far right keys and with Z either above or below Q or J:

Rank: Layout Score

1:  J U O X D I E S G K Y A V L W B H T N C R M F P ' , - " . ? Q Z    0.604749435469232
1:  J U O X D I E S G K Y A V L W B H T N C R M F P ' , - " . ? Z Q    0.6047493417658528
1:  Q U O X D I E S G K Y A V L W B H T N C R M F P ' , - " . ? Z J    0.6047473486481894
1:  Q U O X D I E S G K Y A V L W B H T N C R M F P ' , - " . ? J Z    0.6047471677101812
2:  P F M R C N T H B W L V A Y K G S E I D X O U Q ' , - " . ? Z J    0.6037437839295329
2:  P F M R C N T H B W L V A Y K G S E I D X O U Q ' , - " . ? J Z    0.6037416226754981
2:  P F M R C N T H B W L V A Y K G S E I D X O U J ' , - " . ? Z Q    0.6037413895700938
2:  P F M R C N T H B W L V A Y K G S E I D X O U J ' , - " . ? Q Z    0.603735008585236

We will choose the third layout above, so that Q and U are together
```

**Comment by binarybottle** [comment, marked as answer]
Thank you for asking.  Please disregard this exercise, which has been superceded by the most recent work I've been doing.

**Comment by vleher** [comment]
Version 3 does look quite different from the previous one. How close is ver 3 to completion?

**Reply by malcolmbastien** [reply]
Everybody's eagerly waiting and following the latest commits 😄

**Comment by binarybottle** [comment]
Thank you for asking!  I am running some extra tests to reign in upper movements of the index and little fingers.  We're close!

**Reply by binarybottle** [reply]
@sunaku, @vleher, and @malcolmbastien -- You will be happy to hear that I have settled on an official Engram layout -- version 2!:

https://github.com/binarybottle/engram/issues/8#issuecomment-832437281

**Reply by sunaku** [reply]
Hooray! :tada: Thank you for your herculean efforts on this research! :100::sparkles::bow:  I've been eagerly following its evolution and am glad that the various other permutations have been given worthy consideration and that the rigorous methodology you have since developed for Engram 2.5 and 3.0 is now going to be applied toward Engram 2.0's analysis & explanation. :ok_hand:

**Reply by binarybottle** [reply]
Thank you so much for your interest and enthusiastic support! 

Next, I will update the layout comparison notebook to continue evaluating v2...

**Comment by binarybottle** [comment]
Please see https://github.com/binarybottle/engram-2021/discussions/86#discussion-10126586 for an announcement of the new Engram approach/layouts/repo!
